### ISSUE TITLE
Target URL passing in new authentication system

### ISSUE THREAD

**Issue by tburry** [vanilla/vanilla#7399]
There seem to be some issues with target URLs being passed through the new /authenticate pages.

## Support targets in GET /api/v2/authenticate/authenticators

The URLs returned in the `ui.url` field should all accept a `target` querystring parameter that will be applied correctly to the sign in round trip.

## Support targets in /authenticate/signin

The `?target=` qs parameter should be supported on this page. This means that it is added to the SSO buttons and is respected 

## Support targets in /authenticate/signin/:authenticatorType/:authenticatorID

The `?target=` qs parameter should be supported on this page and passed along the redirect. On initial inspection it looks as though this is being done for Vanilla Connect.

## Support targets in each authenticator that currently works on /authenticate/signin

There are currently authenticators or stub authenticators for the following:

- Facebook
- Twitter
- Google
- Vanilla Connect
- Password - This is a special case, but targets should work for regular passwords.

## Support targets in POST /api/v2/authenticate

- This endpoint MUST return the `targetUrl` field upon successful authentication. It can return It can return '/'.
- Each authenticator must decode its target and pass it through to the result. We cannot rely on the the `?target=` querystring for this since each authenticator has rules on where it's data is gathered.
- I think this means that target URLs must be supported in the `SSOData` class.

## Support targets in POST /api/v2/link-user

- This endpoint MUST return the `targetUrl` field upon successful authentication. It can return '/'.

## Add an /authenticate/redirect?target= page

- This page should redirect to the target.
- The URL should be checked with `isSafeUrl()`. If the URL is safe then the redirect can take place, otherwise redirect to `/`.
- Redirects after authentication should be redirected though this page rather than directly. This means that the final redirect to the target should be done through this, not the redirects to initiate authentication.

## Support targets in /authenticate/connect

- This page needs to be updated to redirect through `/authenticate/redirect`.

## Support targets in the LinkUser components

Successful link users need to be updated to redirect through `/authenticate/redirect`.

## TL:DR

This should be everything. However I understand the basic point that the ?target= parameter should properly work when signing in. Do some QA!

**Comment by donovandb** [comment]
Hello, please see:
[This thread](https://open.vanillaforums.com/discussion/36498/vanilla-2-6-logout-target-single-sign-out)

Is this related?   If so, is there a patch that I can apply to my 6.2 install?  I need this to work as we will be migrating a few SSO Vanilla 2.3 forums to Vanilla 2.6 soon. 
Thx.
